pkg/plugin/oauth2/setup.go

```diff
@@ -1,14 +1,27 @@
 package oauth2
 
 import (
+	"net/url"
+
+	"github.com/hellofresh/janus/pkg/config"
 	"github.com/hellofresh/janus/pkg/jwt"
-	"github.com/hellofresh/janus/pkg/oauth"
+	"github.com/hellofresh/janus/pkg/notifier"
 	"github.com/hellofresh/janus/pkg/plugin"
 	"github.com/hellofresh/janus/pkg/proxy"
+	"github.com/hellofresh/janus/pkg/router"
+	"github.com/pkg/errors"
 	log "github.com/sirupsen/logrus"
 )
 
+var (
+	repo        Repository
+	adminRouter router.Router
+)
+
 func init() {
+	plugin.RegisterEventHook(plugin.StartupEvent, onStartup)
```

Comment:
Not really sure yet if this is better, but maybe we could get rid of the globals above with the following design:
```go
oauth2Plugin := &OAuth2Plugin{repo: repo}
plugin.RegisterEventHook(plugin.StartupEvent, oauth2Plugin.onStartup)
plugin.RegisterEventHook(plugin.ReloadEvent, oauth2Plugin.onReload)
```